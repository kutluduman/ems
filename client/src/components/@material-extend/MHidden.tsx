import { ReactElement } from "react";
import { Breakpoint, Theme, useMediaQuery } from "@mui/material";

export enum Width {
  xsDown = "xsDown",
  smDown = "smDown",
  mdDown = "mdDown",
  lgDown = "lgDown",
  xlDown = "xlDown",
  xsUp = "xsUp",
  smUp = "smUp",
  mdUp = "mdUp",
  lgUp = "lgUp",
  xlUp = "xlUp",
}

interface MHiddenProps {
  width: Width;
  children?: React.ReactNode;
}

const MHidden = ({
  width,
  children,
}: MHiddenProps): ReactElement<any> => {
  const breakpoint = width.substring(0, 2) as Breakpoint;

  const hiddenUp = useMediaQuery((theme: Theme) =>
    theme.breakpoints.up(breakpoint)
  );
  const hiddenDown = useMediaQuery((theme: Theme) =>
    theme.breakpoints.down(breakpoint)
  );

  if (width.includes("Down")) {
    return (hiddenDown ? null : children) as ReactElement;
  }

  if (width.includes("Up")) {
    return (hiddenUp ? null : children) as ReactElement;
  }

  return null as never as ReactElement;
}

export default MHidden;
