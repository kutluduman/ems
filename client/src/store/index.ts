import { configureStore, ThunkAction, Action } from "@reduxjs/toolkit";
import auth from "./slices/auth";
import employee from "./slices/employee";

export const store = configureStore({
  reducer: {
    auth,
    employee,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;