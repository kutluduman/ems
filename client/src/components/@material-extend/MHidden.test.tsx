import { create } from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";

import ThemeConfig from "../../theme";
import { store } from "../../store";
import MHidden, { Width } from "./MHidden";

describe("MHidden Component", () => {
  test("renders the MHidden Component", () => {
    const tree = create(
      <Provider store={store}>
        <Router>
          <ThemeConfig>
            <MHidden width={Width.lgDown}>
              <h1>Some Content</h1>
            </MHidden>
          </ThemeConfig>
        </Router>
      </Provider>
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});