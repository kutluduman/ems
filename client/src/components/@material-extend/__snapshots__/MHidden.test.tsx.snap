// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MHidden Component renders the MHidden Component 1`] = `
<h1>
  Some Content
</h1>
`;
